File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        map<int,int>mpp;//{el,idx}
        for(int i=0;i<n;i++){
            int temp=target-nums[i];
            if(mpp.find(temp)!=mpp.end()){ //found in map
                return {i,mpp[temp]};
            }
            mpp[nums[i]]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>mpp=new HashMap<>();

        for(int i=0;i<nums.length;i++){
            int curr=nums[i];
            int rem=target-curr;

            if(mpp.containsKey(rem)==true){
                int[] arr = new int[2];
                arr[0]=mpp.get(rem);
                arr[1]=i;

                return arr;
            }

            mpp.put(curr,i);
        }

        return new int[0];
    }
}